nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/KeyValueStoreOverflowRecordFilter.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * (C) Copyright 2019 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+package org.nuxeo.ecm.core.work;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.kv.KeyValueService;
+import org.nuxeo.runtime.kv.KeyValueStore;
+
+/**
+ * Filter that use a KeyValueStore to pass big record value
+ *
+ * @since 11.1
+ */
+public class KeyValueStoreOverflowRecordFilter extends BaseOverflowRecordFilter {
```

Comment:
do we really need both implementations ?

Reply:
Both impl are interesting, KV is fast (and has no GC overhead when using Redis) it should be a good choice if the big value are < 10m.
The TS impl is more for XXL value.